modules/lfs/http_client.go

```diff
@@ -134,71 +137,84 @@ func (c *HTTPClient) performOperation(ctx context.Context, objects []Pointer, dc
 		return fmt.Errorf("TransferAdapter not found: %s", result.Transfer)
 	}
 
+	errGroup, groupCtx := errgroup.WithContext(context.Background())
```

Comment:
it should use `ctx`

Reply:
I did initially, but that didn't work because I think the parent context is not backgrounded, so you still only get single-threaded behavior unless you do this

Reply:
It should use the correct parent ctx, otherwise the "background ctx" makes the operation out of control and can't be canceled.

Reply:
you're right, this was a bug in my integration testing, I've put this back to inheriting the parent context - thanks for the callout.